Update channel shuffle to return alias instead of self as-is

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99745
* #99181

Partially addresses https://github.com/pytorch/pytorch/issues/99655

soulitzer committed:
Update channel shuffle to return alias instead of self as-is

[ghstack-poisoned]

jbschlosser reviewed (commented):
Change LGTM, thanks! might be worth renaming the `test_channel_shuffle_return_self` test slightly + its documentation

albanD reviewed (approved):
SGTM
Agreed with Joel on test name

soulitzer committed:
Update on "Update channel shuffle to return alias instead of self as-is"


Partially addresses https://github.com/pytorch/pytorch/issues/99655

[ghstack-poisoned]

soulitzer commented:
@pytorchbot merge
